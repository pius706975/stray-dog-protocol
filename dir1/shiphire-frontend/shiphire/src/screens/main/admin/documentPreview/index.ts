export {default as AdminDocumentPreview} from './AdminDocumentPreview';
