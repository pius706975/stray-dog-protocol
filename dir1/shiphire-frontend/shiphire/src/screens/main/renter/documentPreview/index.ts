export {default as RenterDocumentPreview} from './RenterDocumentPreview';
