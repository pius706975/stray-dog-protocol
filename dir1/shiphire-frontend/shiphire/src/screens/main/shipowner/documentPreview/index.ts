export {default as OwnerDocumentPreview} from './OwnerDocumentPreview';
