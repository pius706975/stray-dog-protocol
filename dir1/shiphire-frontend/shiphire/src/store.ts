import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
