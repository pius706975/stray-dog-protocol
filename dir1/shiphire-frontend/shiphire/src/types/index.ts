export * from './Components'
export * from './Navigators'
export * from './ReduxToolkit'
export * from './Screens'
export * from './Services'