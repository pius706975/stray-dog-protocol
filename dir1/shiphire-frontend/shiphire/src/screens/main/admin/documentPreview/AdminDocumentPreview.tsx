import React from 'react';
import { useTranslation } from 'react-i18next';
import { View } from 'react-native-ui-lib';
import PDFView from 'react-native-view-pdf';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import { Color } from '../../../../configs';
import { modalSlice } from '../../../../slices';
import { AdminDocumentPreviewProps } from '../../../../types';

const AdminDocumentPreview: React.FC<AdminDocumentPreviewProps> = ({
    route,
}) => {
    const dispatch = useDispatch();
    const { hideModal, showModal } = modalSlice.actions;
    const { documentUrl, isButtonActive, btnText } = route.params;
    const { t } = useTranslation('common');

    const handleError = () => {
        dispatch(
            showModal({
                status: 'failed',
                text: t('DocumentPreview.failedLoad'),
            }),
        );
        setTimeout(() => {
            dispatch(hideModal());
        }, 3000);
    };

    return (
        <>
            <ScreenLayout testId="DocumentShipScreen" backgroundColor="light">
                {documentUrl && (
                    <PDFView
                        fadeInDuration={250.0}
                        style={{
                            flex: 1,
                            borderWidth: 1,
                            backgroundColor: Color.bgColor,
                        }}
                        resource={documentUrl}
                        resourceType={'url'}
                        onError={error => handleError()}
                    />
                )}
            </ScreenLayout>
            {isButtonActive && (
                <View
                    style={{
                        position: 'absolute',
                        bottom: 30,
                        right: 30,
                    }}>
                    <Button
                        title={btnText ? btnText : ''}
                        color="success"
                        onSubmit={() => {
                            // onClick!!();
                        }}
                    />
                </View>
            )}
        </>
    );
};

export default AdminDocumentPreview;
