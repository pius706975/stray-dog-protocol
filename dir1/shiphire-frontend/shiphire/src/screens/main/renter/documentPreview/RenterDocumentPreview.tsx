import React from 'react';
import { useTranslation } from 'react-i18next';
import { View } from 'react-native-ui-lib';
import PDFView from 'react-native-view-pdf';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import { Color } from '../../../../configs';
import { modalSlice } from '../../../../slices';
import { RenterDocumentPreviewProps } from '../../../../types';

const RenterDocumentPreview: React.FC<RenterDocumentPreviewProps> = ({
    route,
}) => {
    const dispatch = useDispatch();
    const { t } = useTranslation('common');
    const { hideModal, showModal } = modalSlice.actions;
    const { documentUrl, isEditable } = route.params;

    const handleError = () => {
        dispatch(
            showModal({
                status: 'failed',
                text: t('RenterDocPreview.failedLoad'),
            }),
        );
        setTimeout(() => {
            dispatch(hideModal());
        }, 3000);
    };

    return (
        <>
            <ScreenLayout testId="DocumentShipScreen" backgroundColor="light">
                {documentUrl && (
                    <PDFView
                        fadeInDuration={250.0}
                        style={{
                            flex: 1,
                            borderWidth: 1,
                            backgroundColor: Color.bgColor,
                        }}
                        resource={documentUrl}
                        resourceType={'url'}
                        onError={error => handleError()}
                    />
                )}
            </ScreenLayout>
            {isEditable && (
                <View
                    style={{
                        position: 'absolute',
                        bottom: 30,
                        right: 30,
                    }}>
                    <Button
                        title="Edit Document"
                        color="success"
                        onSubmit={() => {}}
                    />
                </View>
            )}
        </>
    );
};

export default RenterDocumentPreview;
