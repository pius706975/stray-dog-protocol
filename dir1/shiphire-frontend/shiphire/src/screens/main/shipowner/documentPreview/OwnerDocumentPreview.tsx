import React from 'react';
import { View } from 'react-native-ui-lib';
import PDFView from 'react-native-view-pdf';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import { Color } from '../../../../configs';
import { modalSlice, progressIndicatorSlice } from '../../../../slices';
import { AcceptRFQRequest, OwnerDocumentPreviewProps } from '../../../../types';
import { useTranslation } from 'react-i18next';
import { useUpdateTransactionAcceptRFQ } from '../../../../hooks';

const OwnerDocumentPreview: React.FC<OwnerDocumentPreviewProps> = ({
    navigation,
    route,
}) => {
    const { documentUrl, shipId, documentName } = route.params;
    const dispatch = useDispatch();
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;
    const { hideModal, showModal } = modalSlice.actions;
    const mutationUpdateTransactionAcceptRFQ = useUpdateTransactionAcceptRFQ();
    const { t } = useTranslation('common');

    const handleError = () => {
        dispatch(
            showModal({
                status: 'failed',
                text: t('DocumentPreview.failedLoad'),
            }),
        );
        setTimeout(() => {
            dispatch(hideModal());
        }, 3000);
    };

    const handleEditDocument = () => {
        navigation.navigate('EditDocumentForm', {
            documentUrl: documentUrl,
            shipId: shipId!!,
            documentName,
        });
    };

    return (
        <>
            <ScreenLayout testId="DocumentShipScreen" backgroundColor="light">
                {documentUrl && (
                    <PDFView
                        fadeInDuration={250.0}
                        style={{
                            flex: 1,
                            borderWidth: 1,
                            backgroundColor: Color.bgColor,
                        }}
                        resource={documentUrl}
                        resourceType={'url'}
                        onError={error => handleError()}
                    />
                )}
            </ScreenLayout>

            {/* <View
                style={{
                    position: 'absolute',
                    bottom: 30,
                    right: 30,
                }}>
                <Button
                    title="Edit Document"
                    color="success"
                    onSubmit={() => {
                        handleEditDocument();
                    }}
                />
            </View> */}
        </>
    );
};

export default OwnerDocumentPreview;
