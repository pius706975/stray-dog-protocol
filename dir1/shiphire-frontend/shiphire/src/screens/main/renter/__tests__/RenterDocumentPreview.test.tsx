import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MainStackParamList } from '../../../../types';
import MockAdapter from 'axios-mock-adapter';
import httpRequest from '../../../../services/api';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import PDFView from 'react-native-view-pdf';
import { RenterDocumentPreview } from '../documentPreview';
import { render, waitFor } from '@testing-library/react-native';

jest.mock('react-native-view-pdf');

const queryClient = new QueryClient();
const MainStack = createNativeStackNavigator<MainStackParamList>();
const mockAdapter = new MockAdapter(httpRequest);

const mockScreen = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainStack.Navigator>
                    <MainStack.Screen
                        name="RenterDocumentPreview"
                        component={RenterDocumentPreview}
                        initialParams={{
                            documentUrl: 'mock document url',
                            isEditable: true,
                        }}
                    />
                </MainStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

describe('Testing document preview screen', () => {
    describe('Snapshot Testing', () => {
        it('should render document preview screen correctly', () => {
            const docPreviewScreen = render(mockScreen);
            expect(docPreviewScreen).toMatchSnapshot();
        });
    });
    describe('Unit Testing', () => {
        it('should show an alert when fail show document', async () => {
            render(mockScreen);

            await waitFor(() => {
                const modalState = store.getState().modal;

                expect(modalState).toEqual({
                    visible: true,
                    status: 'failed',
                    text: 'RenterDocPreview.failedLoad',
                });
            });
        });
    });
});
