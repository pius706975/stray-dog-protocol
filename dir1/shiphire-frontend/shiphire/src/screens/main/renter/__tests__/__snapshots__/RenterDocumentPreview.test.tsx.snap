// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing document preview screen Snapshot Testing should render document preview screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="DocumentShipScreen"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": undefined,
                  "paddingVertical": undefined,
                }
              }
            >
              <View />
            </RCTScrollView>
          </View>
          <View
            style={
              [
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "bottom": 30,
                  "position": "absolute",
                  "right": 30,
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#1fcc44",
                  "borderRadius": 6,
                  "elevation": 2,
                  "flexDirection": "row",
                  "gap": 0,
                  "justifyContent": "center",
                  "paddingHorizontal": 10,
                  "paddingVertical": 6,
                }
              }
            >
              <Text
                numberOfLines={0}
                style={
                  {
                    "borderBottomWidth": 0,
                    "borderColor": undefined,
                    "color": "#f0f6fb",
                    "fontFamily": "Kanit-Medium",
                    "fontSize": 18,
                    "lineHeight": undefined,
                    "textAlign": undefined,
                  }
                }
              >
                Edit Document
              </Text>
              <View
                style={
                  [
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    {
                      "position": "absolute",
                      "right": 20,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="RenterDocumentPreview"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
